dace/transformation/passes/simplification/control_flow_raising.py

```diff
@@ -139,10 +150,33 @@ def _lift_conditionals(self, sdfg: SDFG) -> int:
                     graph.add_edge(block, conditional, InterstateEdge())
 
                     # Populate branches.
+                    full_cond_expression: Optional[sympy.Basic] = None
+                    uncond_generated = False
                     for i, oe in enumerate(oedges):
                         branch_name = 'branch_' + str(i) + '_' + block.label
                         branch = ControlFlowRegion(branch_name, sdfg)
-                        conditional.add_branch(oe.data.condition, branch)
+
+                        if not oe.data.is_unconditional():
+                            if i == len(oedges) - 1 and oe.data.condition_sympy() == sympy.Not(full_cond_expression):
+                                if uncond_generated:
+                                    warnings.warn(
+                                        f'Control flow raising: Found multiple unconditional branches in {block.label}')
+                                uncond_generated = True
+                                cond = None
+                            else:
+                                cond = oe.data.condition
+                                if full_cond_expression is None:
+                                    full_cond_expression = oe.data.condition_sympy()
+                                else:
+                                    full_cond_expression = sympy.And(full_cond_expression, oe.data.condition_sympy())
```

Comment:
Unsure if this always works (sometimes it evaluates and simplifies wrongly), I'd rather use ast.BoolOp or something